main.js

```diff
@@ -1 +1,78 @@
+import { startDrawingLine, stopDrawingLine, sketchLine } from './path'
+import { startDrawingRect, stopDrawingRect, sketchRect } from './rectangle'
+import { startDrawingCircle, stopDrawingCircle, sketchCircle } from './circle'
+import { canvas, ctx } from './canvas'
+import { drawShapes } from './utils'
+
+const pencil = document.querySelector('.toolbar__pencil')
+const rectangle = document.querySelector('.toolbar__rectangle')
+const circle = document.querySelector('.toolbar__circle')
+const undo = document.querySelector('.toolbar__undo')
+let shapes = []
+let selectedTool = null
+
+canvas.addEventListener('mousedown', event => {
+  switch (selectedTool) {
+    case 'pencil':
+      return startDrawingLine(event)
+    case 'rect':
+      return startDrawingRect(event)
+    case 'circle':
+      return startDrawingCircle(event)
+    default:
+      return null
+  }
+})
+
+canvas.addEventListener('mouseup', () => {
+  switch (selectedTool) {
+    case 'pencil':
+      return stopDrawingLine(path => {
+        shapes = shapes.concat(path)
```

Comment:
why are you not drawing here line? 🤔

Reply:
it's not neccessary bacause drawing lines does not require clearCanvas()